tests/router.test.js

```diff
@@ -81,12 +107,158 @@ test('Login route when logging with correct password is successful', t => {
         .expect(302)
         .expect('Content-Type', 'text/plain; charset=utf-8')
         .end((err, res) => {
+          t.error(err, 'No error');
           t.equal(res.statusCode, 302, 'Status code is 302 for redirecting');
           t.equal(res.header['location'], 'home', 'Should redirect to home page if password is correct');
           t.end();
         });
     });
 });
+test('GET authenticated routes', (t) => {
+  const authenticatedPages = [
+    'about',
+    'home',
+    'symptoms',
+    'appointments',
+    'background',
+    'progress',
+    'info-page',
+    'colour_scheme',
+    'send',
+    'finish',
+  ];
+  t.plan(authenticatedPages.length*4);
+  // First login to get cookie
+  request(app)
+    .post('/login')
+    .type('form')
+    .send({ email: 'jam@gmail.com', password: 'password' })
+    .end((getCookieErr, loginRes) => {
+      const cookies = loginRes.headers['set-cookie'];
+      authenticatedPages.forEach((page) => {
+        // Then make requests to each authenticated route
+        // setting cookie for each request
+        request(app)
+          .get(`/${page}`)
+          .set('Cookie', cookies)
+          .expect(200)
+          .end((err, res) => {
+            t.equal(res.statusCode, 200, `${page} responds with 200`);
+            t.ok(res.text.includes('All about me'), `${page} route renders ${page} page, containing \'All about me\' text`);
+            t.notok(res.redirect, `${page} doesn\t redirect`);
+            t.error(err, 'No error');
+          });
+      });
+    });
+});
+
+test('POST authenticated routes (form sections)', (t) => {
+  const authenticatedPages = [
+    'about',
+    'appointments',
+    'background',
+    'symptoms',
+  ];
+  t.plan(authenticatedPages.length*3);
+  // First login to get cookie
+  request(app)
+    .post('/login')
+    .type('form')
+    .send({ email: 'jam@gmail.com', password: 'password' })
+    .end((getCookieErr, loginRes) => {
+      const cookies = loginRes.headers['set-cookie'];
+      authenticatedPages.forEach((page) => {
+        // Then make requests to each authenticated route
+        // setting cookie for each request
+        request(app)
+          .post(`/${page}`)
+          .set('Cookie', cookies)
+          .expect(302)
+          .end((err, res) => {
+            t.equal(res.statusCode, 302, `${page} responds with 302 (redirects)`);
+            t.ok(res.redirect, `${page} redirects`);
+            t.error(err, 'No error');
+          });
+      });
+    });
+});
+test('POST LOGOUT authenticated routes (form sections)', (t) => {
```

Comment:
Could this test and the following test be combined into one by combining both of the `authenticatedPages` arrays? I couldn't see any need for them to be separate but i'm not sure

Reply:
I probably can, but because the form sections have post functions, it allowed adding extra tests, rather than the rest of them who simply redirect so that's mainly the reason.

Reply:
ah ok cool :)